bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandUnitDiscoveryService.java

```diff
@@ -0,0 +1,230 @@
+            JsonArray jsonArray = jsonParser.parse(response).getAsJsonArray();
+            if (jsonArray.isJsonArray()) {
+                try {
+                    for (JsonElement unit : jsonArray) {
+                        TouchWandUnitData touchWandUnit;
+                        touchWandUnit = TouchWandUnitFromJson.parseResponse(unit.getAsJsonObject());
+                        if (touchWandUnit == null) {
+                            continue;
+                        }
+                        if (!touchWandBridgeHandler.isAddSecondaryControllerUnits()) {
+                            if (!Arrays.asList(CONNECTIVITY_OPTIONS).contains(touchWandUnit.getConnectivity())) {
+                                continue;
+                            }
+                        }
+                        String type = touchWandUnit.getType();
+                        if (!Arrays.asList(SUPPORTED_TOUCHWAND_TYPES).contains(type)) {
+                            logger.debug("Unit discovery skipping unsupported unit type : {} ", type);
+                            continue;
+                        }
+                        switch (type) {
+                            case TYPE_WALLCONTROLLER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_WALLCONTROLLER);
+                                break;
+                            case TYPE_SWITCH:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_SWITCH);
+                                notifyListeners(touchWandUnit);
+                                break;
+                            case TYPE_DIMMER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_DIMMER);
+                                notifyListeners(touchWandUnit);
+                                break;
+                            case TYPE_SHUTTER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_SHUTTER);
+                                break;
+                            default:
+                                continue;
+                        }
+                    }
+                } catch (JsonSyntaxException e) {
+                    logger.warn("Could not parse unit {}", e.getMessage());
+                }
+            }
+        } catch (JsonSyntaxException msg) {
+            logger.warn("Could not parse list units response {}", msg.getMessage());
+        }
+    }
+
+    private void notifyListeners(TouchWandUnitData touchWandUnit) {
+        /* sometimes current status received null , no point update listeners */
+        if (touchWandUnit.getCurrStatus() == null) {
+            return;
+        }
+
+        for (TouchWandUnitStatusUpdateListener listener : listeners) {
+            listener.onDataReceived(touchWandUnit);
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
```

Comment:
Call super destructor at the end.
```suggestion
        removeOlderResults(getTimestampOfLastScan());
        super.stopScan();
```